### PR TITLE
Support for four byte AS Numbers

### PR DESCRIPTION
**Pull request by pashukl**
Summary of Changes:
    - Added valid ASN range : as numbers are 4-byte unsigned integers.  This change will support the right range for AS numbers while peering with customers.
    - Corresponding changes will also be done in SDK and powershell.

Related Bug Id: #1404213

This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [ ] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ ] My spec meets the review criteria:
  - [ ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [ ] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

File: specification/network/resource-manager/Microsoft.Network/2017-10-01/expressRouteCircuit.json [hunk 1]
```diff
@@ -1162,7 +1162,9 @@
         },
         "peerASN": {
           "type": "integer",
-          "format": "int32",
+          "format": "int64",
```
Comment (jhendrixMSFT) [inline specification/network/resource-manager/Microsoft.Network/2017-10-01/expressRouteCircuit.json]: If these numbers are four bytes why the change to int64?  I addition this will introduce a breaking change in SDKs.

**Review by jhendrixMSFT — CHANGES_REQUESTED** [review]

**Comment by pashukl** [comment]
The json integer is not bounded by any range.
However, the format specifier is used to bound the values.

The swagger does not support  “uint32” format.  Hence the option is to use next bigger format which is “int64”.
Additionally, the minimum and maximum have been added so that valid ranges can be handled.

**Review by jhendrixMSFT — APPROVED** [review]
